### DISCUSSION TITLE
How to handle a stripe 100% off coupon?

### DISCUSSION THREAD

**Discussion by hobbypunk90** [pay-rails/pay#966] [Q&A]
Hello,
we are not sure, how to handle an stripe 100% coupon. for subscriptions it works, but charges don't through the lack of a payment intent on the checkout.session.completed event.
Greets,
Marcel

**Comment by excid3** [comment]
I don't believe Stripe PaymentIntents support coupons. https://docs.stripe.com/api/payment_intents/create

You can use Stripe Checkout though and it should support applying a coupon to a one-time charge.
